File: apps/modernization-api/src/main/java/gov/cdc/nbs/patient/profile/investigation/PatientInvestigation.java
package gov.cdc.nbs.patient.profile.investigation;

import java.time.LocalDate;

record PatientInvestigation(
    long investigation,
    LocalDate startedOn,
    String condition,
    String status,
    String caseStatus,
    String jurisdiction,
    String event,
    String coInfection,
    String notification,
    String investigator,
    boolean comparable
) {
}

File: apps/modernization-ui/src/pages/patient/profile/investigation/PatientInvestigation.ts
export enum Headers {
    StartDate = 'Start date',
    Condition = 'Condition',
    Status = 'Status',
    CaseStatus = 'Case status',
    Notification = 'Notification',
    Jurisdiction = 'Jurisdiction',
    Investigator = 'Investigator',
    Investigation = 'Investigation #',
    CoInfection = 'Co-infection #'
}

export type Investigation = {
    investigation: string;
    startedOn?: Date | null;
    condition: string;
    status: string;
    caseStatus?: string | null;
    jurisdiction: string;
    event: string;
    coInfection?: string | null;
    notification?: string | null;
    investigator?: string | null;
    comparable: boolean;
};
